src/pages/PassportRedemption/SignIn/uploadScreen.tsx

```diff
@@ -199,10 +214,19 @@ const UploadScreen = () => {
       <InputContainer className="bottom">
         <Button
           primary={true}
-          disabled={!participatingSeller || !receipt || Number(billTotal) < 10}
+          disabled={
+            !allParticipatingSellers.find(
+              (seller) => seller.name === participatingSeller
+            ) ||
```

Comment:
i dont rlly like the way i wrote this check (this is to ensure that the inputted seller is an actual participating seller). but my issue when thinking about this was: initial call for all participating sellers comes back as an array, so i would have to do either a o(n) call and store all sellers in a map initially or i would have to do a o(n) check here. 

i chose to put it here for now since the other would require more refactoring, but idk is there something else i could do? lmk thoughts!!

Reply:
hmmm, i would say make a [disabled, setDisabled] state, and then in a handleChange button ,we could call setDisabled and if the instant that we receive a proper participating seller in the input, set the disabled prop to false. But for now this is good